aspnet/mvc/views/view-components.rst

```diff
@@ -18,57 +27,91 @@ New to ASP.NET MVC 6, view components are similar to partial views, but they are
+- Parameters come from the calling method, not HTTP, there is no model binding
+- Are not reachable directly as an HTTP endpoint, they are invoked from your code (usually in a view). A view component never handles a request
+- Are overloaded on the signature rather than any details from the current HTTP request
+
+View search path
+^^^^^^^^^^^^^^^^^^
+
+The runtime searches for the view in the following paths:
+
+  - Views/<controller_name>/Components/<view_component_name>/<view_name>
+  - Views/Shared/Components/<view_component_name>/<view_name>
+  
+The default view name for a view component is *Default*, which means your view file will typically be named *Default.html*. You can specify a different view name when creating the view component result or when calling the ``View`` method.
+ 
+We recommend you name the view file *Default.cshtml* and use the *Views/Shared/Components/<view_component_name>/<view_name>* path. The ``PriorityList`` view component used in this sample uses *Views/Shared/Components/PriorityList/Default.cshtml* for the view component view.
+
+Invoking a view component
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To use the view component, call ``@Component.InvokeAsync("Name of view component", <parameters>)`` from a view. The parameters will be passed to the ``InvokeAsync`` method.  The ``PriorityList`` view component developed in the article is invoked from the *Views/Todo/Index.cshtml* view file. In the following, the ``InvokeAsync`` method is called with two parameters:
+
+.. code-block:: HTML
+
+   @await Component.InvokeAsync("PriorityList", 2, false)
+   
+Invoking a view component directly from a controller
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+View components are typically invoked from a view, but you can invoke them directly from a controller method. While view components do not define endpoints like controllers, you can easily implement a controller action that returns the content of a `ViewComponentResult <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/ViewComponentResult/index.html>`__. 
 
-    using System.Linq;
-    using Microsoft.AspNet.Mvc;
-    using TodoList.Models;
+In this example, the view component is called directly from the controller:
 
-    namespace TodoList.ViewComponents
-    {
-      public class PriorityListViewComponent : ViewComponent
-      {
-        private readonly ApplicationDbContext db;
+.. literalinclude:: view-components/sample/ViewCompFinal/Controllers/ToDoController.cs
+ :language: c#
+ :lines: 23-26
+ :dedent: 6
 
-        public PriorityListViewComponent(ApplicationDbContext context)
-        {
-          db = context;
-        }
+Download the starter sample
```

Comment:
Can we capture the entire walk-through section under a single heading? Maybe something like "Walkthrough: Creating a Simple View Component"?